File: LuaMeadow/DWMediator.cpp
//
//  DWMediator.cpp
//  LuaMeadow
//
//  Created by Larusso on 25.08.13.
//  Copyright (c) 2013 Larusso. All rights reserved.
//

#include "DWMediator.h"

DWMediator::DWMediator()
{
    std::cout << "++ ------- new DWMediator" << std::endl;
}

DWMediator::~DWMediator()
{
    std::cout << "++ ------- destroy DWMediator" << std::endl;
}

void DWMediator::release()
{

}

void DWMediator::printType()
{
    std::cout << "++ ------- printType called" << std::endl;
}

File: LuaMeadowiOS/DWMediator.m
//
//  DWMediator.m
//  LuaMeadow
//
//  Created by Larusso on 28.08.13.
//  Copyright (c) 2013 Larusso. All rights reserved.
//

#import "DWMediator.h"

@implementation DWMediator

void * Mediator()
{
    return (__bridge void *)([[DWMediator alloc]init]);
}

- (void) release_me
{
    NSLog(@"----OBJ-C-> release called");
}

- (void) printType
{
    NSLog(@"----OBJ-C-> printType called");
}

- (void) move:(int) x andY:(int) y
{
    NSLog(@"----OBJ-C-> move %d,%d called",x,y);
}

- (void) doSomething
{
    NSLog(@"----OBJ-C-> doSomething called");
}
@end
